provider/cmd/app/app.go

```diff
@@ -455,3 +464,37 @@ func runStreamingServer(ctx context.Context, setupLog, log logr.Logger, srv *ser
 	}
 	return nil
 }
+
+func runMetricsServer(ctx context.Context, setupLog logr.Logger, addr string) error {
+	setupLog.Info("Starting metrics server on " + addr)
+
+	mux := http.NewServeMux()
+	mux.Handle("/metrics", promhttp.Handler())
+
+	srv := http.Server{
+		Addr:         addr,
+		Handler:      mux,
+		ReadTimeout:  5 * time.Second,
```

Comment:
Is it a good idea to make these timeouts configurable?

Reply:
it is good idea, but i did it for streaming server and maybe grpc too. So what do you think about follow up issue?

Reply:
It appears that there are no timeout parameters specified in the gRPC server. Additionally, in the current implementations, timeout settings are not defined for the streaming server, which implies that there will never be a timeout. The question is, are we considering introducing timeouts for the streaming server?

I would suggest making timeouts for the metrics server here and creating issues for other components if we plan to do so.

Reply:
GRPC has timeout: https://pkg.go.dev/google.golang.org/grpc#ConnectionTimeout

timeouts are for protection of our app. because you can do attacks with long living connections. 5 *time.Second is extremely long time.
but i will do it

Reply:
fixed